File: Problems/0458-poor-pigs.py
class Solution(object):
    def poorPigs(self, buckets, minutesToDie, minutesToTest):
        if buckets == 1:
            return 0

        states = minutesToTest // minutesToDie + 1
        pigs = 0
        power = 1

        while power < buckets:
            pigs += 1
            power *= states

        return pigs

File: Problems/0458-poor-pigs.cpp
#include <bits/stdc++.h>
using namespace std;

// LeetCode solution starts from here
class Solution {
public:
    int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        if (buckets == 1)
            return 0;

        int states = minutesToTest / minutesToDie + 1;
        int pigs = 0;
        long long power = 1;

        while (power < buckets) {
            pigs++;
            power *= states;
        }

        return pigs;
    }
};